go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/operator-framework/operator-sdk
 
-go 1.13
+go 1.15
```

Comment:
We cannot ask for users to use 1.15 to use the tool and 1.13 to work with their project built via the tool.  
We need to do this change for all which means that we need first change kb/upstream than bump here the kb/commit with the change.

Reply:
@camilamacedo86  Can you clarify what you mean by "cannot" and "need"?

Go 1.15 should be fully backwards compatible with Go 1.13, so there should not be a technical reason that SDK is forced to use the minimum Go version of all of its dependencies.

I _can_ see an argument that it might be confusing to Operator SDK contributors that the project is compiled with 1.15 buts scaffolds projects compatible with 1.13 (_users_ should not see any difference whatsoever).

But to me at least, that is not a strong enough reason to block SDK bumping to a newer go version that has lots of nice improvements.

Reply:
Hi @joelanford,
 
If it is ONLY for contributor then would not change the end-user docs and add fragment.
And in this way, I agree that is technically possible. However, see the complexities raise with this change: 

- CI will use go 1.15 to test the projects built and we are no longer using the version officially supported. 
- It brings confusion as you highlight above
- Note that, if I not wrong we have issues raised were the root cause of the problem was users are using go 1.15 instead of the 1.13 go version supported. I mean, that was solved by users changing their env to use the go 1.13 instead of 1.15. And then, to test the projects built with the tool locally and work in the issues will not be required to install/uninstall the go version to switch between 1.13 and 1.15? So, cannot it bring an unnecessary effort?  
- Do we really win value by doing it before changing the go version for the scaffolded projects instead of doing together as it was done in the past? 
 
IHMO: Would be better to do together and avoid the above scenarios. However, since it passed in the e2e tests I am ok with your final decision.

Reply:
> CI will use go 1.15 to test the projects built and we are no longer using the version officially supported.

It will use a go1.15 toolchain, but the projects will still specify go1.13, which tells the Go toolchain what to enable or disable. See https://golang.org/ref/mod#go-mod-file-go. I think this is reasonable.

> Note that, if I not wrong we have issues raised were the root cause of the problem was users are using go 1.15 instead of the 1.13 go version supported. I mean, that was solved by users changing their env to use the go 1.13 instead of 1.15. And then, to test the projects built with the tool locally and work in the issues in some scenarios at least will not be required to install/uninstall the go version to switch between 1.13 and 1.15? So, cannot it bring an unnecessary effort?

I'm not sure if that's entirely the case. I think some (maybe all) of those were related to GOPROXY and the way some distributions set the default in their distro repo packages vs the default that is used when you install from https://golang.org/dl/. I would not be surprised if users who switched go versions to solve their problem uninstalled their distro package and installed from Go's download site, thus incidentally solving the actual problem by picking up a different default for GOPROXY.